src/IRMatch.cpp

```diff
@@ -283,4 +283,126 @@ bool expr_match(Expr pattern, Expr expr, map<string, Expr> &matches) {
     }
 }
 
+namespace IRMatcher {
+
+HALIDE_ALWAYS_INLINE
```

Comment:
Yes, if you don't then the objects passed around have to actually exist somewhere, and the stack blows up. Absolutely everything needs to be force-inlined to prevent this.